syft/source/source.go

```diff
@@ -304,6 +307,73 @@ func NewFromImage(img *image.Image, userImageStr string) (Source, error) {
 	}, nil
 }
 
+func (s *Source) ID() artifact.ID {
+	return s.id
+}
+
+func (s *Source) SetID() {
```

Comment:
Should this PR include a call to this function?

Reply:
^ Second PR is going to do that so we can link it into relationships - Just wanted to keep this PR small, but I can add the call so we're setting it on construction

Reply:
Looking again, there also isn't ONE great place to put the single call since we have functions like `New`, `NewFromImage` `NewFromDirectory` `NewFromFile`.

Let me see all the places it needs to be included so we have full coverage of all the source creation

Reply:
I could just have it get set on the first call to `ID` if s.id = ""